pkg/printers/humanreadable.go

```diff
@@ -250,23 +250,20 @@ func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) er
 			// If this table has column definitions, remember them for future use.
 			h.lastColumns = table.ColumnDefinitions
 		}
-		if len(h.options.ExtraColumns) != 0 {
-			// Initialize JSON parsers
-			columns := h.options.ExtraColumns
-			parsers := make([]*jsonpath.JSONPath, len(columns))
-			for ix := range columns {
-				parsers[ix] = jsonpath.New(fmt.Sprintf("extracolumn%d", ix)).AllowMissingKeys(true)
-				if err := parsers[ix].Parse(columns[ix].FieldSpec); err != nil {
-					return err
-				}
+		columns := h.options.ExtraColumns
```

Comment:
I'd expect this work similar to DecorateTable below by injecting the additional user-requested columns. It will make the code more readable and avoid printing logic duplication.